### DISCUSSION TITLE
MYSQL_RANDOM_ROOT_PASSWORD=yes

### DISCUSSION THREAD

**Discussion by eals2019** [firefly-iii/firefly-iii#7856] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

MariaDB root credentials for FireFly_III DB. I'm trying to restore MySQLDump, but the firefly user doesn't have enough privileges.

### Debug information

Here's what I've been doing... I've created a crontab that will dump FireFly's database using mysqldump, then upload it to Google Drive using RClone. I got everything to work, but I can't successfully restore from the mysqldump. I always run into issues like _Syntax error_ or _Access Denied_. 

Here's my thought-flow for how I've been attempting this... If you see I'm doing something wrong, PLEASE correct me. I know I'm using this section of the report incorrectly, but bear with me.

Attempting to restore from host machine (not docker container):
-
**$ docker exec -i firefly_iii_db mariadb -u firefly -pPassword123 < /home/elewis/GoogleDrive/Docker-Containers/FireFly-Backups/FireFly_Dump-08.15.2023-20.20.01.sql**
_ERROR 1064 (42000) at line 1: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'OCI runtime exec failed: exec failed: unable to start container process: exec...' at line 1_

Attempting to restore from inside container (after copying MySQLDump to container successfully):
-
**# mariadb -u firefly -pProverbs19:17! firefly_iii < /home/FireFly_Dump-08.15.2023-20.20.01.sql**
_ERROR 1044 (42000): Access denied for user 'firefly'@'%' to database 'firefly_iii'_

My last attempt was to use the database root user. I learned that in the **.db.env** document, there is a variable **MYSQL_RANDOM_ROOT_PASSWORD=yes**, but this would naturally mean that I don't know the random root password. So how would I find this root password, or how would I adjust the **.db.env** file to specify my own root password instead of the random root password (I would prefer to specify... I know it's less secure, but this is home setup). I'm thinking that the appropriate user permissions would allow me to restore DB from the mysqldump.

It is possible that I'm missing the obvious means of backing up firefly... copying the FireFly docker volumes. Let me know if that's what I should be doing.

**Comment by JC5** [comment]
You can Google how to adjust the settings for the MariaDB container: https://hub.docker.com/_/mariadb

- Remove `MYSQL_RANDOM_ROOT_PASSWORD` entirely.
- Set `MARIADB_ROOT_PASSWORD=secret`

If you remove the db volume (otherwise nothing will change) and start again, you should end up with a database with your chosen root password.
